### PR TITLE
pkg/github will no longer set GITHUB_TOKEN variable

### PR DESCRIPTION
**Pull request by wilsonehusin**
Signed-off-by: Wilson E. Husin <whusin@vmware.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Previously, NewWithToken and NewEnterpriseWithToken will assign
token to GITHUB_TOKEN, then calls New / NewEnterprise which expects
token from environment variable

This PR inverts the behavior such that New / NewEnterprise will
attempt to look for token from GITHUB_TOKEN environment variable, then
calls NewWithToken / NewEnterprise which takes the token through
parameter

The change is to present a predictable behavior for anyone importing /
vendoring pkg/github -- consumer may be loading tokens with other safe
methods (config file, etc) not involving environment variable and would
be surprising if they find out their process / child process suddenly
exposes the secret to environment variable

#### Which issue(s) this PR fixes:

Partially in theme of #1694

#### Special notes for your reviewer:

This change _should_ be backwards compatible with prior implementation

This code inspired [the fun Twitter thread](https://twitter.com/wilsonehusin/status/1359345956982906881)

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by wilsonehusin** [comment]
I was on the edge on submitting this PR [but then I took it to Twitter first](https://twitter.com/wilsonehusin/status/1359345956982906881). Thanks for chatting!

File: pkg/github/github.go [hunk 1]
```diff
@@ -127,13 +127,31 @@ type Client interface {
 	) (*github.IssueComment, *github.Response, error)
 }
 
+// TODO: we should clean up the functions listed below and agree on the same
+// return type (with or without error):
+// - New
+// - NewWithToken
+// - NewEnterprise
+// - NewEnterpriseWithToken
```
Comment (wilsonehusin) [inline pkg/github/github.go:135]: I put this as `TODO` for backwards compatibility reasons. Are we okay with breaking API? I know that at least [zeitgeist](https://github.com/kubernetes-sigs/zeitgeist) imports `pkg/github`.

File: pkg/github/github.go [hunk 2]
```diff
@@ -127,13 +127,31 @@ type Client interface {
...
 // environment variable will result in an authenticated client.
 // If the $GITHUB_TOKEN is not set, then the client will do unauthenticated
 // GitHub requests.
 func New() *GitHub {
-	ctx := context.Background()
 	token := util.EnvDefault(TokenEnvKey, "")
+	client, err := NewWithToken(token)
+	if err != nil {
+		return nil
+	}
```
Comment (wilsonehusin) [inline pkg/github/github.go:146]: yeah this doesn't really make sense considering NewWithToken never really returns `error` -- doing it to make `errcheck` linter happy.

Comment (cpanato) [reply pkg/github/github.go:146]: maybe we ignore that and add a nolint comment?

```
client, _ := NewWithToken(token) // nolint: errcheck
```

Comment (wilsonehusin) [reply pkg/github/github.go:146]: sure, I'm not too opinionated on either approaches 😄 I'll switch it to nolint if nobody says anything today

Comment (wilsonehusin) [reply pkg/github/github.go:146]: eh whatever I'll do it now lol

**Comment by wilsonehusin** [comment]
Bump -- checking if we have more thoughts about this 😄

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**